src/main/java/com/artipie/asto/Splitting.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto;
+
+import io.reactivex.Flowable;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import org.reactivestreams.Publisher;
+
+/**
+ * Splits the original ByteBuffer to several ones
+ * with size less or equals defined max size.
+ *
+ * @since 1.12.0
+ */
+public class Splitting {
+
+    /**
+     * Source byte buffer.
+     */
+    private final ByteBuffer source;
+
+    /**
+     * Max size of split byte buffer.
+     */
+    private final int size;
+
+    /**
+     * Ctor.
+     *
+     * @param source Source byte buffer.
+     * @param size Max size of split byte buffer.
+     */
+    public Splitting(final ByteBuffer source, final int size) {
+        this.source = source;
+        this.size = size;
+    }
+
+    /**
+     * Splits the original ByteBuffer to ones with size less
+     * or equals defined max {@code size}.
+     *
+     * @return Publisher of ByteBuffers.
+     */
+    public Publisher<ByteBuffer> publisher() {
+        final Publisher<ByteBuffer> res;
+        int remaining = this.source.remaining();
+        if (remaining > this.size) {
+            final List<ByteBuffer> parts = new ArrayList<>(16);
```

Comment:
why 16? It seems, like we can calculate the size by dividing `remaining` into `this.size`